src/test/java/com/puppycrawl/tools/checkstyle/XDocsPagesTest.java

```diff
@@ -244,4 +267,230 @@ private static void testCheckstyleXml(String fileName, String code, String unser
                     + unserializedSource);
         }
     }
+
+    @Test
+    public void testAllCheckSections() throws IOException, CheckstyleException,
+            ParserConfigurationException {
+        final ClassLoader cl = XDocsPagesTest.class.getClassLoader();
+        final Set<String> packageNames = PackageNamesLoader.getPackageNames(cl);
+        final ModuleFactory moduleFactory = new PackageObjectFactory(packageNames, cl);
+
+        for (File file : Files.fileTreeTraverser().preOrderTraversal(XDOCS_DIRECTORY)) {
+            final String fileName = file.getName();
+
+            if (!fileName.startsWith("config_") || "config_reporting.xml".equals(fileName)) {
+                continue;
+            }
+
+            final String input = Files.toString(file, UTF_8);
+            final Document document = getRawXml(fileName, input, input);
+            final NodeList sources = document.getElementsByTagName("section");
+
+            for (int position = 0; position < sources.getLength(); position++) {
+                final Node section = sources.item(position);
+                final String sectionName = section.getAttributes().getNamedItem("name")
+                        .getNodeValue();
+
+                if ("Content".equals(sectionName) || "Overview".equals(sectionName)) {
+                    continue;
+                }
+
+                System.out.println(fileName + " section '" + sectionName + "'");
+
+                Assert.assertTrue(fileName + " section '" + sectionName
+                        + "' shouldn't end with 'Check'", !sectionName.endsWith("Check"));
+
+                Object clss = null;
+
+                try {
+                    clss = moduleFactory.createModule(sectionName);
+                }
+                catch (CheckstyleException e) {
+                    Assert.fail(fileName + " couldn't find class: " + sectionName);
+                }
+
+                int subSectionPos = 0;
+                for (Node subSection : getChildrenElements(section)) {
+                    final String subSectionName = subSection.getAttributes().getNamedItem("name")
+                            .getNodeValue();
+
+                    // can be in different orders, and completely optional
+                    if (subSectionName.equals("Notes"))
+                        continue;
+                    if (subSectionName.equals("Rule Description"))
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/checkstyle/checkstyle/file/1183653445/issues/source?bid=2193633&fileBranchId=2921952#l320)